module/os/linux/zfs/arc_os.c

```diff
@@ -174,84 +159,84 @@ arc_evictable_memory(void)
 static unsigned long
 arc_shrinker_count(struct shrinker *shrink, struct shrink_control *sc)
 {
-	return (btop((int64_t)arc_evictable_memory()));
+	/*
+	 * kswapd doesn't know how much we evict, because it's only looking
+	 * for pages to be added to the inactive lists.  This causes it to
+	 * ask us to evict the entire ARC.  Instead, we ignore its requests
+	 * and manage the free memory in arc_reap_cb[_check]().
+	 */
+	if (current_is_kswapd()) {
+		/*
+		 * Wake up the arc_evict_zthr so it can start responding to
+		 * this memory pressure right away, if it isn't already.
+		 */
+		arc_wait_for_eviction(0);
+		return (0);
+	}
+
+	/*
+	 * __GFP_FS won't be set if we are called from ZFS code.  To avoid a
```

Comment:
How about referencing `kmem_flags_convert()` here as a pointer to further reading to help explain why this is true for ZFS code sections wrapped with `spl_fstrans_mark()/unmark()`.